### PR TITLE
Add support for ai_bots_protection field in the cloudflare bot manage…

### PR DESCRIPTION
**Pull request by gcloudflare**
**Can only be reviewed after cloudflare-go dependency bumped to v0.104.0.**

Added support for ai_bots_protection in the cloudflare bot management resource.  This setting was added to cloudflare-go in [this PR](https://github.com/cloudflare/cloudflare-go/pull/2974).

### CODE REVIEW

**Comment by jacobbednarz** [comment]
v0.104.0 has landed and i've just bumped your branch to get the latest changes.

**Comment by gcloudflare** [comment]
https://github.com/cloudflare/cloudflare-go/releases/tag/v0.104.0 introduced some breaking changes in dns.go. I will revisit this PR once that is merged.

**Comment by jacobbednarz** [comment]
the underlying DNS issue has been fixed and your branch rebased. looks like you have type issues in the tests but the rest is looking good.

File: internal/sdkv2provider/resource_cloudflare_bot_management_test.go [hunk 1]
```diff
@@ -90,12 +94,13 @@ func testCloudflareBotManagementSBFM(resourceName, rnd string, bm cloudflare.Bot
 		sbfm_likely_automated = "%[5]s"
 		sbfm_verified_bots = "%[6]s"
 		sbfm_static_resource_protection = "%[7]t"
-		optimize_wordpress = "%[8]t"
+		optimize_wordpress = "%[8]t",
+		ai_bots_protection = "%[9]t"
```
Comment (jacobbednarz) [inline internal/sdkv2provider/resource_cloudflare_bot_management_test.go:98]:
```suggestion
		ai_bots_protection = "%[9]s"
```

File: internal/sdkv2provider/resource_cloudflare_bot_management_test.go [hunk 2]
```diff
@@ -106,8 +111,9 @@ func testCloudflareBotManagementEntSubscription(resourceName, rnd string, bm clo
 		enable_js = "%[3]t"
 
 		suppress_session_score = "%[4]t"
-		auto_update_model = "%[5]t"
+		auto_update_model = "%[5]t",
+		ai_bots_protection = '%[6]t'
```
Comment (jacobbednarz) [inline internal/sdkv2provider/resource_cloudflare_bot_management_test.go:115]:
```suggestion
		ai_bots_protection = '%[6]s'
```

File: internal/sdkv2provider/resource_cloudflare_bot_management_test.go [hunk 3]
```diff
@@ -90,12 +94,13 @@ func testCloudflareBotManagementSBFM(resourceName, rnd string, bm cloudflare.Bot
 		sbfm_likely_automated = "%[5]s"
 		sbfm_verified_bots = "%[6]s"
 		sbfm_static_resource_protection = "%[7]t"
-		optimize_wordpress = "%[8]t"
+		optimize_wordpress = "%[8]t",
```
Comment (jacobbednarz) [inline internal/sdkv2provider/resource_cloudflare_bot_management_test.go:97]:
```suggestion
		optimize_wordpress = "%[8]t"
```

File: internal/sdkv2provider/resource_cloudflare_bot_management_test.go [hunk 4]
```diff
@@ -106,8 +111,9 @@ func testCloudflareBotManagementEntSubscription(resourceName, rnd string, bm clo
 		enable_js = "%[3]t"
 
 		suppress_session_score = "%[4]t"
-		auto_update_model = "%[5]t"
+		auto_update_model = "%[5]t",
```
Comment (jacobbednarz) [inline internal/sdkv2provider/resource_cloudflare_bot_management_test.go:114]:
```suggestion
		auto_update_model = "%[5]t"
```

File: internal/sdkv2provider/resource_cloudflare_bot_management_test.go [hunk 5]
```diff
@@ -107,7 +112,8 @@ func testCloudflareBotManagementEntSubscription(resourceName, rnd string, bm clo
 
 		suppress_session_score = "%[4]t"
 		auto_update_model = "%[5]t"
+		ai_bots_protection = '%[6]s'
```
Comment (jacobbednarz) [inline internal/sdkv2provider/resource_cloudflare_bot_management_test.go:115]:
```suggestion
		ai_bots_protection = "%[6]s"
```

**Comment by jacobbednarz** [comment]
acceptance tests all passing

```
TF_ACC=1 go test ./internal/sdkv2provider -v -run "^TestAccCloudflareBotManagement_" -count 1 -timeout 120m -parallel 1
=== RUN   TestAccCloudflareBotManagement_SBFM
--- PASS: TestAccCloudflareBotManagement_SBFM (3.83s)
=== RUN   TestAccCloudflareBotManagement_Unentitled
--- PASS: TestAccCloudflareBotManagement_Unentitled (0.84s)
PASS
ok  	github.com/cloudflare/terraform-provider-cloudflare/internal/sdkv2provider	5.850s
```

**Pull request merged**